packages/open-truss/src/configuration/engine-v1/RenderConfig.tsx

```diff
@@ -111,9 +121,17 @@ function createSignals(signalsConfig: SignalsV1, validate: boolean): Signals {
     }
 
     if (signal) {
-      const s = signal.parse(undefined)
-      s.yamlName = name
-      signals[name] = s
+      if (name in initialSignalValues) {
+        const initialValue = initialSignalValues[name]
+        const s = signal.parse(undefined)
+        s.yamlName = name
+        s.value = initialValue
+        signals[name] = s
+      } else {
+        const s = signal.parse(undefined)
+        s.yamlName = name
+        signals[name] = s
+      }
```

Comment:
I think this should also work and is a bit easier to read:

```suggestion
      const s = signal.parse(undefined)
      s.yamlName = name
      signals[name] = s

      if (name in initialSignalValues) {
        s.value = initialSignalValues[name]
      }
```